### DISCUSSION TITLE
Any way to receive events from tsparticles?

### DISCUSSION THREAD

**Discussion by htopia** [matteobruni/tsparticles#2468] [Q&A] [answered]
Hi there,
Is there a way to subscribe or receive events about the emitters of tsparticles?
For example, I would like to know if an emitter has finished based on duration, or maybe subscribe to a progress event of an emitter so I can trigger an action in my code based on its progress.

I'm using react-tsparticles so I was picturing something like:

```
<Particles
    height="100%" width="100%"
    options={myOptions}
    onEmitterEnd={(emitterName) => console.log(emitterName+" has finished")}
/>
```

**Comment by matteobruni** [comment, marked as answer]
Hi @htopia,

I was thinking about events, but it's not a thing yet.

You can access the emitter plugin instance using this code: `container.plugins.get("emitters")` (`container` is the parameter returned in the `loaded` event or using the `container` ref property of the `<Particles />` component. There's an `array` property containing all the emitters loaded, maybe it could help for now.
